base/gcutils.jl

```diff
@@ -116,6 +116,17 @@ another Task or thread.
 """
 enable_finalizers(on::Bool) = on ? enable_finalizers() : disable_finalizers()
 
+"""
+    GC.take_heap_snapshot(io::IOStream)
+
+Write a snapshot of the heap, in the JSON format expected by the Chrome
+Devtools Heap Snapshot viewer (.heapsnapshot extension), to the given
+IO stream.
+"""
+function take_heap_snapshot(io)
```

Comment:
```suggestion
function take_heap_snapshot(io::IOStream)
```

Reply:
tried that earlier, but it seems not to be defined yet:

```
$ make
...
LoadError(at "sysimg.jl" line 3: LoadError(at "Base.jl" line 97: LoadError(at "gcutils.jl" line 119: UndefVarError(var=:IOStream))))
```

Reply:
looks like this made the builds fail, e.g. https://build.julialang.org/#/builders/63/builds/3778/steps/8/logs/stdio

either we need to get IOStream in scope somehow, or take the annotation back off

Reply:
taking the annotation back off until we can figure out how to get it in scope in this file… I guess we have to figure out what file `IOStream` is defined in and make sure it gets included before this

Reply:
still may be good to acquire the internal lock on `io` here first also